jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -308,7 +320,11 @@ def _get_defaults_from_metadata(self, dataset=None):
             # if display unit is different, translate
             if (self.config == 'cubeviz') and (self.display_unit != ''):
                 disp_unit = u.Unit(self.display_unit)
-                mjy2abmag = (mjy2abmag * u.Unit("MJy/sr")).to_value(disp_unit)
+                mjy2abmag = flux_conversion_general(mjy2abmag,
+                                                    u.Unit("MJy/sr"),

```

Comment:
This would save the overhead of string parsing.
```suggestion
                                                    u.MJy / u.sr,
```